servicetalk-data-protobuf-jersey/src/main/java/io/servicetalk/data/protobuf/jersey/ProtobufMediaTypes.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2022 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.data.protobuf.jersey;
+
+import io.servicetalk.concurrent.api.Publisher;
+import io.servicetalk.http.api.HttpHeaderValues;
+
+import javax.ws.rs.core.MediaType;
+
+/**
+ * Media type definitions for Protobuf.
+ */
+public final class ProtobufMediaTypes {
+    /**
+     * {@code "application/x-protobuf"}.
+     * @see #APPLICATION_PROTOBUF_TYPE
+     * @see HttpHeaderValues#APPLICATION_PROTOBUF
+     */
+    public static final String APPLICATION_PROTOBUF = "application/x-protobuf";
+    /**
+     * {@code "application/x-protobuf-var-int"}
+     * <a href="https://developers.google.com/protocol-buffers/docs/encoding">base 128 VarInt protobuf encoding</a>.
+     * Should be used with {@link Publisher} to stream data.
+     * @see #APPLICATION_PROTOBUF_VAR_INT_TYPE
+     * @see HttpHeaderValues#APPLICATION_PROTOBUF_VAR_INT
+     */
+    public static final String APPLICATION_PROTOBUF_VAR_INT = "application/x-protobuf-var-int";
```

Comment:
I couldn't find a canonical content type to use in this scenario, the motivation is when using streaming the encoding (var-int) is different than scalar (raw single object). the different maybe subtle for people, but at least we should have exception messages to indicate the content type isn't compatible if people try to mix them.

@chemicL @bondolo - have you come across this scenario in spring use cases?

Reply:
I believe that what's done here is the correct approach, a different mime-type. There doesn't appear to be any standard for the delimited format. I find it kind of strange that Google recommends the approach but doesn't attempt to standardize it since presumably it will be commonly used.